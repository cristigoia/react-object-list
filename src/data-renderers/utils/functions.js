/**
 * Gets the value of the cell from the row and keys provided
 * @param  {Object} row  Row supplied
 * @param  {Array|string} keys List of keys used to access the value
 * @return {(Object|string|number)}      Value from row, or row itself if no keys
 */
const getValueFromAccessor = (row, keys) => {
  if (!Array.isArray(keys)) keys = keys.split('.')
  let value = row
  const cloneKeys = [...keys]
  while (value && cloneKeys.length) {
    const thisKey = cloneKeys.shift()
    if (thisKey in value) {
      value = value[thisKey]
    }
  }
  return value
}

export {
  getValueFromAccessor,
}
