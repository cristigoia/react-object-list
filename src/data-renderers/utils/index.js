export {
  getValueFromAccessor,
  handleRowClick,
} from './functions'
