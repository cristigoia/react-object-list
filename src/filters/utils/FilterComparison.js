import React from 'react'
import PropTypes from 'prop-types'
import Select from 'react-select'

class FilterComparison extends React.Component {
  static propTypes = {
    options: PropTypes.arrayOf(PropTypes.shape({
      label: PropTypes.string,
      value: Select.propTypes.value,
    })),
    value: Select.propTypes.value,
    onChange: PropTypes.func.isRequired,
  }

  render() {
    const {options, value} = this.props
    return (
      <div className="objectlist-current-filter__filter-comparison">
        <Select
          options={options}
          value={value}
          onChange={newValue => this.props.onChange(newValue.value)}
          clearable={false}
        />
      </div>
    )
  }
}

export default FilterComparison
