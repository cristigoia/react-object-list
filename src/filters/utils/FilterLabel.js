import React from 'react'
import PropTypes from 'prop-types'

/**
 * Component to render label for filters
 */
class FilterLabel extends React.Component {
  static propTypes = {
    /** Text to render in the label */
    label: PropTypes.string.isRequired,
  }
  render() {
    return (
      <label className="objectlist-current-filter__filter-name">{this.props.label}:</label>
    )
  }
}

export default FilterLabel
