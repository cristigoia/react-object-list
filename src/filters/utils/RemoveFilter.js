import React from 'react'
import PropTypes from 'prop-types'

/**
 * Component to render button used to remove a filter
 */
class RemoveFilter extends React.Component {
  static propTypes = {
    /** Function to be called to remove filter */
    onClick: PropTypes.func.isRequired,
  }

  render() {
    return (
      <button className="objectlist-button objectlist-button--delete objectlist-button--icon objectlist-button--borderless" onClick={this.props.onClick} title="Remove Filter" />
    )
  }
}

export default RemoveFilter
