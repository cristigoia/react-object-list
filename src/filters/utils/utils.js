const possibleQueryStringLookupExpressions = ['exact', 'lt', 'gt', 'gte', 'lte']

export {
  possibleQueryStringLookupExpressions,
}
