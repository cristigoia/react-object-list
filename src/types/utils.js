function floatPrecision(floatValue, precision) {
  floatValue = parseFloat(floatValue)
  if (isNaN(floatValue)) { return parseFloat('0').toFixed(precision) } else {
    const power = Math.pow(10, precision)
    floatValue = (Math.round(floatValue * power) / power).toFixed(precision)
    return floatValue.toString()
  }
}

function commaNumber(numberString) {
  const parts = numberString.toString().split('.')
  parts[0] = parts[0].replace(/\B(?=(\d{3})+(?!\d))/g, ',')
  return parts.join('.')
}
/**
 * Converts a float to stringed currency representation
 * @param  {number} floatNumber Amount to convert
 * @return {string}             Dollar string
 */
export function currency(floatNumber) {
  return `${floatNumber < 0 ? '-' : ''}$${commaNumber(floatPrecision(Math.abs(floatNumber), 2))}`
}
