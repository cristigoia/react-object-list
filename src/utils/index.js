export * from './constants'
export * from './proptypes'
export * from './functions'
export loadingSpinner from './loading'
export ErrorMessage from './ErrorMessage'
