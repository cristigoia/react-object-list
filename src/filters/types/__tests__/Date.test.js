import React from 'react'
import Moment from 'moment'
import { shallow } from 'enzyme'
import { snapshotTest } from 'utils/tests'
import DateComponent from '../Date'

jest.mock('react-day-picker/DayPickerInput', () => 'DayPickerInput')
jest.mock('react-select')

describe('DateComponent', () => {
  const baseProps = {
    onChange: jest.fn(),
    value: Date(),
    fixedComparison: {value: 'fixed', label: 'Exact'},
    comparison: 'fixed',
  }
  describe('Snapshots', () => {
    it('renders default', () => {
      snapshotTest(<DateComponent {...baseProps} />)
    })
    it('renders with relative comparison', () => {
      snapshotTest(<DateComponent {...baseProps} comparison="relative" />)
    })
    it('has custom input format', () => {
      snapshotTest(<DateComponent {...baseProps} inputFormat="Do MMM YY" />)
    })
    it('has custom relative date options', () => {
      snapshotTest(<DateComponent {...baseProps} comparison="relative" relativeDateOptions={[
        {value: 'now', label: 'Right now'},
        {value: 'later', label: 'A bit later'},
        {value: 'never', label: 'Never ever'},
      ]} />)
    })
    it('has null value', () => {
      snapshotTest(<DateComponent {...baseProps} value={null} />)
    })
  })
  describe('Functions', () => {
    describe('handles fixed date changing', () => {
      let instance
      beforeEach(() => {
        spyOn(baseProps, 'onChange')
        instance = shallow(<DateComponent {...baseProps} />).instance()
      })
      it('handles valid date', () => {
        const newValue = Moment()
        instance.handleDateValueChange(newValue)
        expect(baseProps.onChange).toHaveBeenCalledWith(newValue)
      })
      it('handles invalid date', () => {
        const newValue = 'Invalid date'
        instance.handleDateValueChange(newValue)
        expect(baseProps.onChange).toHaveBeenCalledWith(null)
      })
    })
    describe('handles relative date changing', () => {
      let instance
      beforeEach(() => {
        spyOn(baseProps, 'onChange')
        instance = shallow(<DateComponent {...baseProps} />).instance()
      })
      it('handles array passed', () => {
        const newValue = ['now']
        instance.handleDateRelativeChange(newValue)
        expect(baseProps.onChange).not.toHaveBeenCalled()
      })
      it('handles single value', () => {
        const newValue = 'now'
        instance.handleDateRelativeChange(newValue)
        expect(baseProps.onChange).toHaveBeenCalledWith(newValue)
      })
    })
  })
})
