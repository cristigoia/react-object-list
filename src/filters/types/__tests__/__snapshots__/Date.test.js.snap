// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date Snapshots has custom input format 1`] = `
<DayPickerInput
  classNames={
    Object {
      "container": "apilist-input__container",
      "overlay": "",
      "overlayWrapper": "apilist-input__daypicker",
    }
  }
  clickUnselectsDay={true}
  dayPickerProps={
    Object {
      "fixedWeeks": true,
    }
  }
  format="DD/MM/YYYY"
  formatDate={[Function]}
  onDayChange={[Function]}
  parseDate={[Function]}
  placeholder="15th Jul 17"
  value="15th Jul 17"
/>
`;

exports[`Date Snapshots has custom relative date options 1`] = `
<Select
  className="apilist-current-filter__active-status"
  clearable={false}
  onChange={[Function]}
  options={
    Array [
      Object {
        "label": "Right now",
        "value": "now",
      },
      Object {
        "label": "A bit later",
        "value": "later",
      },
      Object {
        "label": "Never ever",
        "value": "never",
      },
    ]
  }
  value={"2017-07-15T00:00:00.000Z"}
/>
`;

exports[`Date Snapshots renders default 1`] = `
<DayPickerInput
  classNames={
    Object {
      "container": "apilist-input__container",
      "overlay": "",
      "overlayWrapper": "apilist-input__daypicker",
    }
  }
  clickUnselectsDay={true}
  dayPickerProps={
    Object {
      "fixedWeeks": true,
    }
  }
  format="DD/MM/YYYY"
  formatDate={[Function]}
  onDayChange={[Function]}
  parseDate={[Function]}
  placeholder="15/07/2017"
  value="15/07/2017"
/>
`;

exports[`Date Snapshots renders with relative comparison 1`] = `
<Select
  className="apilist-current-filter__active-status"
  clearable={false}
  onChange={[Function]}
  options={
    Array [
      Object {
        "label": "Today",
        "value": "today",
      },
      Object {
        "label": "Beginning of this week",
        "value": "week_start",
      },
      Object {
        "label": "Beginning of this month",
        "value": "month_start",
      },
      Object {
        "label": "Beginning of this year",
        "value": "year_start",
      },
    ]
  }
  value={"2017-07-15T00:00:00.000Z"}
/>
`;
